gb-dbg/src/debugger/breakpoints/breakpoint.rs

```diff
@@ -49,3 +50,127 @@ impl Breakpoint {
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum Operator {
+    /// Check for equality of value between another, `==`
+    Eq,
+    /// Logical AND, `&&`
+    LogicAnd,
+    /// Logical XOR, `^^`
+    LogicXor,
+    /// Logical OR, `||`
+    LogicOr,
+    /// Binary AND, `&`
+    BinaryAnd,
+    /// Binary XOR, `^`
+    BinaryXor,
+    /// Binary OR, `|`
+    BinaryOr,
+    /// Check for inequality of a value between another, `!=`
+    NotEq,
+    /// Check if a value is greather than another, `>`
+    Sup,
+    /// Check if a value is lower than another, `<`
+    Inf,
+    /// Check if a value is greather or equal to another, `>=`
+    SupEq,
+    /// Check if a value is lower or equal to another, `<=`
+    InfEq,
+}
+
+impl Display for Operator {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Operator::Eq => write!(f, "=="),
+            Operator::NotEq => write!(f, "!="),
+            Operator::Sup => write!(f, ">"),
+            Operator::Inf => write!(f, "<"),
+            Operator::SupEq => write!(f, ">="),
+            Operator::InfEq => write!(f, "<="),
+
+            Operator::BinaryAnd => write!(f, "&"),
+            Operator::BinaryXor => write!(f, "^"),
+            Operator::BinaryOr => write!(f, "|"),
+
+            Operator::LogicAnd => write!(f, "&&"),
+            Operator::LogicXor => write!(f, "^^"),
+            Operator::LogicOr => write!(f, "||"),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Node {
+    Register(CpuRegs),
+    Address(u16),
+    Value(u16),
+    UnaryExpr {
+        op: UnaryOperator,
+        child: Box<Node>,
+    },
+    BinaryExpr {
+        op: Operator,
+        lhs: Box<Node>,
+        rhs: Box<Node>,
+    },
+}
+
+impl Display for Node {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Node::Register(r) => write!(f, "{}", r),
+            Node::Address(addr) => write!(f, "*{:#X}", addr),
+            Node::Value(v) => write!(f, "{:#X}", v),
+            Node::UnaryExpr { op, child } => write!(f, "{}({})", op, child),
+            Node::BinaryExpr { op, lhs, rhs } => write!(f, "{} {} {}", lhs, op, rhs),
+        }
+    }
+}
+
+impl Node {
+    pub fn simple(address: u16) -> Self {
+        Self::BinaryExpr {
+            op: Operator::Eq,
+            lhs: boxed!(Self::Register(CpuRegs::PC)),
+            rhs: boxed!(Self::Value(address)),
+        }
+    }
+}
+
+impl FromStr for Node {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        use nom::combinator::all_consuming;
+        use crate::debugger::breakpoints::parser;
+
+        let (_rest, node) = match all_consuming(parser::expr_complete)(s) {
+            Ok(ret) => ret,
+            Err(e) => {
+                return Err(anyhow!("Invalid input: {}", e));
+            }
+        };
+        Ok(node)
+    }
+}
```

Comment:
duplicate of the content of `parser.rs`

Reply:
i moved it?

Reply:
It's allready existing in the file `parser.rs`, remove the duplicated content in `breakpoint.rs`

Reply:
i don't see any duplicated code ?

Reply:
my bad, I didn't see the commit 2c55d38ce7b1420bc2d53c41db08012d8541d968